internal/sink/sink.go

```diff
@@ -71,12 +85,43 @@ func NewSink(c Config, logger *zap.Logger, listener ancla.Register) Sink {
 		return v1
 	case *ancla.RegistryV2:
 		if len(l.Registration.Webhooks) > 0 {
-			var whs Webhooks
-			for _, wh := range l.Registration.Webhooks {
-				v1 := &WebhookV1{}
-				v1.Update(c, logger, wh.ReceiverURLs[1:], l.GetId(), l.Registration.FailureURL, wh.ReceiverURLs[0])
-				whs = append(whs, v1)
+			var whs WebhookSink
+			r := &HashRing{}
+			whs.HashField = l.Registration.Hash.Field
+			for i, wh := range l.Registration.Webhooks {
+				v2 := &WebhookV2{
+					CommonWebhook: CommonWebhook{
+						id:               l.GetId(),
+						logger:           logger,
+						failureUrl:       l.Registration.FailureURL,
+						deliveryInterval: c.DeliveryInterval,
+						deliveryRetries:  c.DeliveryRetries,
+					},
+				}
+				if len(wh.ReceiverURLs) > 0 {
+					urlCount, err := getUrls(wh.ReceiverURLs[1:])
+					if err != nil {
+						v2.logger.Error("error recevied parsing urls", zap.Error(err))
+					}
+					v2.updateUrls(urlCount, wh.ReceiverURLs[0], wh.ReceiverURLs[1:])
+				} else {
+					sortBy := wh.DNSSrvRecord.LoadBalancingScheme
+					for _, domain := range wh.DNSSrvRecord.FQDNs {
+						dialer := new(Dialer)
+						err := dialer.lookup(domain, sortBy)
+						if err != nil {
+							v2.logger.Error("error received looking up service records", zap.Error(err))
+						}
+						dialer.NewClient()
```

Comment:
do we want to create the client here? or does it make more sense to move to the `send()` function?

Reply:
Here is good, I think I'll push up some updates to help you with the structure. (I'll slack you)